nanoFramework.Logging/LoggerExtensions.cs

```diff
@@ -410,7 +410,11 @@ public static void Log(this ILogger logger, LogLevel logLevel, EventId eventId,
                 throw new ArgumentNullException(nameof(logger));
             }
 
-            logger.Log(logLevel, eventId, string.Format(message, args), exception, MessageFormatter);
+            string formatedMessage = message; 
+            if(args.Length > 0) 
+                formatedMessage = string.Format(message, args);
```

Comment:
We're using explicit brackets in nanoFramework. 

 ```suggestion
       {
                formatedMessage = string.Format(message, args);
        }
```

Reply:
Other wise, you can even do like:
```csharp
string formatedMessage = args.Length > 0 ? string.Format(message, args) : message;
```

Reply:
The shortest solution with explicit brackets would be this:

![2022-12-30_13-04-09](https://user-images.githubusercontent.com/146932/210088283-6096f106-2403-424a-8888-6bfe8148d97f.png)
It would change the reference from the method declaration without using a new variable. On my old PC and ESP32 over Com-Port and with enabled "step filtering messages"  in the debugger, it takes hours for the debugger to jump to the next block of code. There are small facts that are far more important than visual style, which is technically irrelevant.
I personally find it much more important how the code behaves in the debugger. And there is a big difference between if/else and ?: . Even if you break a block with ternary operator, the debugger can't. It may be possible sometime in the future, but it's not possible at the moment.

![2022-12-23_09-53-18](https://user-images.githubusercontent.com/146932/210088310-1dd98430-11d7-42e1-b13b-5568e0d21069.png)

![2022-12-23_09-58-53](https://user-images.githubusercontent.com/146932/210088337-1bf05d5a-1f57-49ed-b219-73d480bfacd0.png)

![2022-12-23_09-54-39](https://user-images.githubusercontent.com/146932/210088348-4e1c5553-164d-4e16-b77c-ef1384c2b638.png)

![2022-12-23_09-56-13](https://user-images.githubusercontent.com/146932/210088363-3d099299-37b1-4271-a06d-7ec00f9b1282.png)

But the most important thing is unfortunately missing from this review.
It's something that can be discussed endlessly, but my feeling about it is very clear:
The code must be documented! And the right place is in the code. No one would know what this code is for unless it is described. Of course it's also good if there is a test that checks this and a requirement call for it and description somewhere in the source control and in a Spec., but you should see it when you look at the code!

PS: The reason why I didn't check in a test is that the test framework doesn't work properly for me. Not even with the newest Christmas versions. It's acting super strange! There is still a lot of work to be done before it works the way it should.